File: javascript/leetcode/single-element-in-a-sorted-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var singleNonDuplicate = function(nums) {
    var previousVal = nums[1]
    
    for (var i = 0; i < nums.length; i++) {
        if (previousVal !== nums[i] && nums[i] !== nums[i+1]) {return nums[i]}
        previousVal = nums[i]
    }
};

File: ruby/leetcode/single-element-in-a-sorted-array.rb
# @param {Integer[]} nums
# @return {Integer}
def single_non_duplicate(nums)
    previous_val = nums[1]
    found_elem = nil
    for i in 0...nums.size()
        if previous_val != nums[i] && (i + 1 == nums.size() || nums[i] != nums[i + 1])
            found_elem = nums[i]
            break
        end
        previous_val = nums[i]
    end
    return found_elem
end
